### PR TITLE
[mlir] Make the ml_program dialect allow all of its operations to be inlined.

### PR DESCRIPTION
**Pull request by stellaraccident**

### CODE REVIEW

**Review by ScottTodd — APPROVED** [review]

**Review by kuhar — COMMENTED** [review]
Not super familiar with this but would this be easy to test? I can see some other dialect have those (LLVM, SPIR-V, Vector, Linalg).

**Comment by stellaraccident** [comment]
> Not super familiar with this but would this be easy to test? I can see some other dialect have those (LLVM, SPIR-V, Vector, Linalg).

Good point. Let me add a test.

**Comment by stellaraccident** [comment]
@kuhar ptal

**Review by kuhar — APPROVED** [review]

File: mlir/lib/Dialect/MLProgram/IR/MLProgramDialect.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 #include "mlir/Dialect/MLProgram/IR/MLProgram.h"
 #include "mlir/IR/DialectImplementation.h"
+#include "mlir/Transforms/InliningUtils.h"
```
Comment (jpienaar) [inline mlir/lib/Dialect/MLProgram/IR/MLProgramDialect.cpp:11]: Why is the utils needed for the interface?

**Pull request merged**